sound/soc/sof/sof-priv.h

```diff
@@ -216,6 +219,10 @@ struct snd_sof_dsp_ops {
 	int (*get_window_offset)(struct snd_sof_dev *sdev,
 				 u32 id);/* mandatory for common loader code */
 
+	/* VirtIO operations */
+	int (*request_topology)(struct snd_sof_dev *sdev, const char *name,
+				struct firmware *fw); /* optional */
```

Comment:
why is getting the topology optional? isn't it pretty much required, and the host provides it?

Reply:
This is the global DSP ops struct. "Normal" hardware implementors (HDA etc.) don't need this operation, for them just request_firmware() is used. Only VirtIO needs it.